src/server/protocol.ts

```diff
@@ -2316,6 +2379,48 @@ namespace ts.server.protocol {
         source?: SymbolDisplayPart[];
     }
 
+    /**
+     * Additional completion entry details, available on demand
+     */
+    export interface RichCompletionEntryDetails {

```

Comment:
Probably not? I just wanted to be explicit with myself when I added the types since I don't work with the protocol much. Now that it's done, I think a union is a fine alternative.